app/src/main/java/org/fossasia/openevent/general/ticket/TicketsRecyclerAdapter.kt

```diff
@@ -26,15 +28,31 @@ class TicketsRecyclerAdapter : RecyclerView.Adapter<TicketViewHolder>() {
         eventCurrency = currencyCode
     }
 
+    fun applyDiscount(discountCode: DiscountCode) {
+        this.discountCode = discountCode
+    }
+
+    fun cancelDiscountCode() {
+        this.discountCode = null
+    }
+
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TicketViewHolder {
         val view = LayoutInflater.from(parent.context).inflate(R.layout.item_ticket, parent, false)
         return TicketViewHolder(view)
     }
 
     override fun onBindViewHolder(holder: TicketViewHolder, position: Int) {
-        val event = tickets[position]
-        holder.bind(event, selectedListener, eventCurrency,
-            if ((this::ticketAndQuantity.isInitialized)) ticketAndQuantity.get(position).second else 0)
+        val ticket = tickets[position]
+        var currentDiscountCode: DiscountCode? = null
+        discountCode?.tickets?.forEach {
+            if (it.id.toInt() == ticket.id)
+                currentDiscountCode = discountCode
+        }
+        val qty = if ((this::ticketAndQuantity.isInitialized)) ticketAndQuantity[position].second else 0
+        if (currentDiscountCode != null)
+            holder.bind(ticket, selectedListener, eventCurrency, qty, currentDiscountCode)
+        else
+            holder.bind(ticket, selectedListener, eventCurrency, qty)
```

Comment:
`currentDiscountCode != null` is unnecessary I think:
 just add `holder.bind(ticket, selectedListener, eventCurrency, qty, currentDiscountCode)`

Reply:
Yes right, that's why I have take null as a default value, forgot to change it. :+1:

Reply:
Updated